nats.go

```diff
@@ -4072,15 +4072,29 @@ func (nc *Conn) PublishRequest(subj, reply string, data []byte) error {
 // Used for handrolled Itoa
 const digits = "0123456789"
 
+// validateSubject checks if the subject contains characters that break the NATS protocol.
+func validateSubject(subj string) error {
+	if subj == "" {
+		return ErrBadSubject
+	}
+	for i := range len(subj) {
+		c := subj[i]
+		if c <= ' ' && (c == ' ' || c == '\t' || c == '\r' || c == '\n') {
```

Comment:
Like the idea of skipping all control characters `c <= ' '`... wonder if just that check is not enough?